source/server/admin/clusters_handler.cc

```diff
@@ -99,6 +99,9 @@ void setHealthFlag(Upstream::Host::HealthFlag flag, const Upstream::Host& host,
     health_status.set_active_hc_timeout(
         host.healthFlagGet(Upstream::Host::HealthFlag::ACTIVE_HC_TIMEOUT));
     break;
+  case Upstream::Host::HealthFlag::EDS_STATUS_DRAINING:
+    health_status.set_eds_health_status(envoy::config::core::v3::DRAINING);
```

Comment:
This is wrong. It should be 
health_status.set_eds_health_status(host.healthFlagGet(Host::HealthFlag::EDS_STATUS_DRAINING));

Reply:
@milton0825 @wbpcode

Reply:
https://github.com/envoyproxy/envoy/pull/30622#discussion_r1518125160 is not correct either as healthFlagGet return a bool and set_eds_health_status expects a health status enum. But it is true that there was a bug introduced with this PR which should get fixed by #32801 .